[Munhui Kim] iP

munhuikim opened the pull request:
# Eli 🤖 Task Manager

> "Automate the routine, so you can focus on the essential." — *Your Digital Assistant* ([source])

**Eli** helps you manage your tasks efficiently, freeing up your mental space for things that truly matter. It's:
- **text-based**
- *easy to learn*
- **FAST and SUPER INTUITIVE** to use

## Getting Started with Eli

All you need to do is:

1. **Download** it from [here](https://github.com/your-repo/releases).
2. **Double-click** the downloaded file.
3. **Add your tasks** to get started.
4. Let **Eli** manage your tasks for you. 🧠

And the best part? It's FREE!

## Features

- [x] Managing tasks
- [x] Searching tasks by keyword
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
 public static void main(String[] args) {
    Eli eli = new Eli("data/eli.txt");
    eli.run();
  }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Level-2

Commit:
Level-3

Commit:
Level-4

Commit:
Level-6

Commit:
no message

Commit:
Level-5

Commit:
Added Command, Ui, Parser and Storage classes

Commit:
Added javadoc

Commit:
UI test files

Commit:
Merge remote-tracking branch 'origin/master' into A-MoreOOP

Commit:
Divide classes into packages

iamdiluxedbutcooler commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,21 @@
+// Delete this file
 public class Duke {
     public static void main(String[] args) {
```
May I ask why you have not deleted the Duke class, given that the new Eli class now manages the main functionality? Having two main classes in the project might lead to confusion among other developers regarding which class should be referenced as the primary one during project setup.

iamdiluxedbutcooler commented on src/main/java/eli/Eli.java:
```diff
@@ -0,0 +1,67 @@
+import eli.command.Command;
+import eli.exception.EliException;
+import eli.parser.Parser;
+import eli.storage.Storage;
```
Got duplicate import statement `eli.storage.Storage;` 😢

iamdiluxedbutcooler commented on src/main/java/eli/Eli.java:
```diff
@@ -0,0 +1,67 @@
+/**
+ * The main class of the Eli application.
+ * Coordinates the interaction between the user, storage, task list, and commands.
+ */
```
Good job on JavaDocs comments!

iamdiluxedbutcooler commented on src/main/java/eli/Eli.java:
```diff
@@ -0,0 +1,67 @@
+   */
+  public Eli(String filePath) {
+    // this.tasks = new TaskList();
+    // this.scanner = new Scanner(System.in);
```
Consider removing unnecessary comments?

iamdiluxedbutcooler commented on src/main/java/eli/command/Command.java:
```diff
@@ -0,0 +1,41 @@
+  protected boolean isExit;
+
+  /**
+   * Constructor for Command.
```
Additional explanation in the commenton why you set `isExit` to false would be beneficial!

iamdiluxedbutcooler commented on src/main/java/eli/command/DeleteCommand.java:
```diff
@@ -0,0 +1,29 @@
+/**
+ * Command to delete a task from the task list.
+ */
+public class DeleteCommand extends Command {
```
LGTM!

iamdiluxedbutcooler commented on src/main/java/eli/parser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+package eli.parser;
+
+import eli.command.*;
```
Imported classes should always be listed explicitly.

iamdiluxedbutcooler commented on src/main/java/eli/storage/Storage.java:
```diff
@@ -0,0 +1,69 @@
+package eli.storage;
+
+import eli.task.*;
```
Imported classes should always be listed explicitly.

iamdiluxedbutcooler reviewed (commented):
Overall LGTM! Covers all functionality and good code quality. Keep up your good work!

Commit:
no message

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
add local date time

Commit:
no message

Commit:
touch up code

Commit:
add junit test

Commit:
add a jar file

Commit:
remove unnecessary comments

Commit:
follow coding standards

Commit:
find a task by searching for a keyword in a task description

DesSnowy commented on src/main/java/eli/Eli.java:
```diff
@@ -0,0 +1,64 @@
+package eli;
+
+import eli.storage.Storage;
+import eli.command.Command;
+import eli.exception.EliException;
+import eli.parser.Parser;
+import eli.storage.Storage;
```
There is duplicate importing of eli.storage.Storage.

DesSnowy commented on src/main/java/eli/command/ListCommand.java:
```diff
@@ -0,0 +1,19 @@
+   * Constructor for ListCommand.
+   */
+  @Override
+  public void execute(TaskList tasks, Ui ui, Storage storage) {
+    tasks.list();
+  }
```
I noticed that other Commands are calling the ui to print out the responses. Would it be better if this was the same as well?

DesSnowy commented on src/main/java/eli/parser/Parser.java:
```diff
@@ -0,0 +1,95 @@
+package eli.parser;
+
+import eli.command.*;
+import eli.exception.*;
+import eli.task.*;
```
Based on the coding standard, we should not import *.

DesSnowy commented on src/main/java/eli/parser/Parser.java:
```diff
@@ -0,0 +1,95 @@
+    String commandWord = parts[0];
+    String description = parts.length > 1 ? parts[1].trim() : "";
+
+    switch (commandWord.toLowerCase()) {
+      case "list":
+        return new ListCommand();
+      case "mark":
+        return new MarkCommand(Integer.parseInt(description));
+      case "unmark":
+        return new UnmarkCommand(Integer.parseInt(description));
+      case "delete":
+        return new DeleteCommand(Integer.parseInt(description));
+      case "todo":
+        if (description.isEmpty()) {
+          throw new EliException("The description of a todo cannot be empty.");
+        }
+        return new AddCommand(new ToDo(description));
+      case "deadline":
+        String[] deadlineParts = description.split(" /by ");
+        if (deadlineParts.length < 2) {
+          throw new EliException("The deadline must be in the format 'description /by yyyy-MM-dd'.");
+        }
+        return new AddCommand(new Deadline(deadlineParts[0], deadlineParts[1]));
+      case "event":
+        String[] eventParts = description.split(" /from ");
+        if (eventParts.length < 2) {
+          throw new EliException("The event must be in the format 'description /from yyyy-MM-dd /to yyyy-MM-dd'.");
+        }
+        String[] times = eventParts[1].split(" /to ");
+        return new AddCommand(new Event(eventParts[0], times[0], times[1]));
+      case "find":
+        if (description.isEmpty()) {
+          throw new EliException("The keyword for finding tasks cannot be empty.");
+        }
+        return new FindCommand(description);
+      case "bye":
+        return new ExitCommand();
+      default:
+        throw new UnknownCommandException();
+    }
```
Based on our coding standard, the indentation for `case` should be on the same level as `switch`.

DesSnowy commented on src/main/java/eli/storage/Storage.java:
```diff
@@ -0,0 +1,68 @@
+
+    try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+      String line;
+      while ((line = reader.readLine()) != null) {
```
Would it be clearer if the assignment to `line` was done inside the while loop?

DesSnowy reviewed (commented):
Good job with the javadocs and naming.

I've pointed out some places that could be refined even more but LGTM otherwise.

Commit:
fix based on coding standard

Commit:
update README.md

Commit:
edit error in README.md

munhuikim committed:
add a gui to the chat bot

munhuikim committed:
delete unnecessary comments

munhuikim committed:
delete eli.jar

munhuikim committed:
Set up JavaFX dependencies in build.gradle

This commit adds the required JavaFX dependencies to the build.gradle
file for the Eli project. The correct JavaFX modules (javafx-controls,
javafx-fxml) have been added to ensure the GUI works as expected.

munhuikim committed:
Add exception handling for invalid task operations

The current implementation of the Eli project does not properly handle
invalid user input when performing task operations like unmarking,
deleting, or marking a task with an invalid index. Currently, an
assertion is used, which is disabled in production environments, causing
unhandled invalid inputs.

To ensure better error handling, I replaced assertions with exceptions
that provide meaningful feedback to the user when invalid operations
occur. Specifically, an IllegalArgumentException is thrown when a task
index is out of bounds.

Changes made:
* Replaced assertion in mark(), unmark(), and delete() methods with
  IllegalArgumentException.
* Updated UnmarkCommand to handle IllegalArgumentException and provide
  user-friendly error messages.
* Added error handling in Ui to display the appropriate error message
  to the user.

This change improves the robustness of the Eli project by handling
unexpected input more gracefully, improving user experience.

munhuikim committed:
Remove unnecessary comments

munhuikim committed:
Refine dialog box design for a more polished look

- Reduced font size in dialog boxes to 14px for a subtler, more refined text display.
- Adjusted dialog box borders to be lighter and thinner (1px) for a more delicate appearance.
- Modified the rounded corners to 15px for a smoother, softer look, aligning with the overall cute theme.
- Applied consistent styling to both user and Eli's dialog boxes to ensure a cohesive interface.

munhuikim committed:
Add Ui.png

munhuikim committed:
no message

munhuikim committed:
Update the given docs\README.md

munhuikim committed:
Change image directory

munhuikim committed:
no message

munhuikim committed:
no message

munhuikim committed:
no message

munhuikim committed:
Update image source

munhuikim committed:
Update user guide

munhuikim committed:
no message

munhuikim committed:
Merge branch 'branch-A-Personality'

# Conflicts:
#	data/eli.txt

munhuikim committed:
Add javadocs for the getResponse method

munhuikim committed:
Added Javadocs for Deadline.java, Tasklist.java, and DialogBox.java
- Documented methods and classes for better code clarity
- Improved maintainability by adding descriptions of parameters, return types, and functionality

munhuikim committed:
Merge branch 'branch-Level-8'

munhuikim committed:
Added Javadocs for Ui class and methods in eli.ui package
- Documented user interaction methods such as readCommand, showWelcome, and showGoodbye
- Provided descriptions for task-related display methods
- Enhanced code readability and maintainability

munhuikim committed:
Refactored code to adhere to coding standards
- Reformatted code for consistency in indentation and spacing
- Improved naming conventions for better clarity
- Ensured proper usage of access modifiers and method declarations
- Enhanced code readability and maintainability by following best practices

munhuikim committed:
Edited FindCommand to search tasks by keyword
- Edited FindCommand class to find tasks containing a specific keyword in their description
- Integrated keyword search functionality with TaskList and Ui for displaying results
- Enhanced task search feature for better user experience

munhuikim committed:
Added GUI to chatbot using JavaFX
- Implemented JavaFX interface for improved user interaction
- Replaced console input/output with graphical components for chat display
- Enhanced user experience with a more intuitive and visually appealing design

munhuikim committed:
Removed redundant comments for cleaner code
- Deleted unnecessary and repetitive comments
- Improved code readability by retaining only meaningful documentation

munhuikim committed:
Updated chatbot messages for improved clarity and engagement
- Refined response phrasing for a more user-friendly interaction
- Enhanced tone and style of messages to better align with chatbot's personality
- Adjusted error and confirmation messages for better readability

munhuikim committed:
Merge pull request #2 from munhuikim/Merging-PRs

Updated chatbot messages for improved clarity and engagement

munhuikim committed:
Add dependencies in build.gradle

munhuikim committed:
Updated Javadocs for improved documentation clarity
- Revised method descriptions for better understanding of functionality
- Added missing parameter and return value explanations
- Enhanced overall readability and consistency of the documentation

munhuikim committed:
Replaced wildcard imports with explicit class imports in Storage.java
- Improved code clarity by specifying individual imports
- Imported necessary classes from java.io, eli.task, eli.exception, and eli.parser packages
- Enhanced maintainability and readability by avoiding wildcard imports

munhuikim committed:
Added Javadocs to Task class and methods
- Documented class-level description for Task, including its role as an abstract base class
- Added detailed explanations for methods such as getStatus, changeDoneStatus, and toFileFormat
- Improved code readability and maintainability with clear documentation of functionality and parameters

munhuikim committed:
Added Javadocs to TaskList class and methods
- Documented class-level overview of TaskList
- Provided detailed descriptions for constructors and methods including addTask, mark, unmark, delete, and findTasksByKeyword
- Enhanced code clarity and maintainability by improving documentation and adding assertions for input validation

munhuikim committed:
Refactored code with minor spacing adjustments and error message reorganization
- Improved code readability by adjusting spacing
- Re-arranged error messages for better clarity and consistency

munhuikim committed:
Update comments

munhuikim committed:
Add README.md

munhuikim committed:
Remove errors in README.md

munhuikim committed:
Remove error in README.md

munhuikim committed:
Ui.java: Change the comment after deleting the task
